system/HTTP/Response.php

```diff
@@ -559,7 +559,7 @@ public function sendBody()
 	 * @return $this
 	 * @throws \CodeIgniter\HTTP\RedirectException
 	 */
-	public function redirect(string $uri, string $method = 'auto', int $code = null)
+	public function redirect(string $uri, $method = 'auto', $code = null)
```

Comment:
Removing the typehints here won't affect anything, as long as there is a default value (even null) it should accept null values just fine. Can you provide example usage that's causing the problem and verify that you're a) on the latest develop commit and b) running under PHP 7+.

Reply:
For the redirect issue, I was simply running the following in my controller after a post request:
```
redirect()->back()->withInput();
```

And getting the following error:
```
TypeError
Argument 2 passed to CodeIgniter\HTTP\Response::redirect() must be of the type string, null given, called in /Library/WebServer/Documents/daylight/FUEL/v2.0_CI/system/HTTP/RedirectResponse.php on line 105	
```
I'm running PHP 7.1.8

Reply:
It looks like the deal is that the `previous_url()` method in the URL helper is returning null. So the solution here doesn't appear to be removing the typehint, but adding a default null value and ensuring things work from there. 

```
redirect(string $uri=null, $method='auto', $code=null);
```